external-crates/move/crates/move-compiler/src/typing/translate.rs

```diff
@@ -3345,6 +3353,10 @@ fn borrow_exp_dotted(context: &mut Context, mut_: bool, ed: ExpDotted) -> Box<T:
                 }
                 exp = Box::new(T::exp(ret_ty, sp(index_loc, e_)));
             }
+            ExpDottedAccess::UnresolvedError => {
+                assert!(idx == num_accessors - 1);
+                break;
```

Comment:
This feels like it will not be enough for the IDE. I would imagine you will need something like `UnannotatedExp_::InvalidAccess(Box<Exp>, Option<Name>)` which will let you float access to invalid fields and invalid methods

Reply:
I am not sure I understand the limitation here and was hoping you could you give an example when this is not going to be sufficient.

I was going for the IDE to be able to understand what all valid prefixes of an otherwise only partially parsed chain are. As per added test it should understand what `s` (in `s.;`) and `s.another_field` (in `s.another_field.;`) are, even if parsing fails after the final dot (as for auto-completion we don't really care what "wrong thing" comes after the final dot)
```
    fun foo(s: AnotherStruct) {
        let _tmp1 = s.;                // incomplete with `;` (next line should parse)
        let _tmp2 = s.another_field.;  // incomplete with `;` (next line should parse)
    }
```

Reply:
If you have `s.foo_` in the IDE, you want all fields and functions that are prefixed by `foo_`

Reply:
For me personally, I never do `s.` and wait, I always _think_ I know the name of the function or field, and then am sadly mistaken. But autocorrect picks up about halfway through me typing.

Reply:
Got it. From what I understand, though, integration with the IDE will work in a slightly different way. There is only one trigger event for auto-completion, and it's the appearance of the `.` character. Whatever you type afterwards does not trigger further auto-completion events. This means, that the LSP implementation needs to provide all possible completions upon just seeing `.` (as otherwise it will not have another chance). It may appear as if it's the LSP that does the filtering (in your example, show only functions prefixed by `foo_`) but it's the "client" side that actually does it.

Reply:
I can include IDE integration in this PR and make sure that everything works as expected but per @cgswords's previous request I am trying to keep the compiler PRs and IDE PRs separate.

Reply:
OK, I take it back :-) I thought it over and now I am understanding much better what you had in mind! More changes (likely) coming!

Reply:
I reworked things a bit. We indeed need new `UnannotatedExp_::InvalidAccess` to support a dot-chain with no proper identifier after the dot (e.g., `tmp.;`), even if the dot is on a separate line (that's why we need to preserve dot's location in conjunction with the type of expression that precedes it, so that the symbolicator can get the preceding expression type having only location of the dot at its disposal).

I don't think we need anything new to represent a dot chain with an "invalid" field or method name name (e.g., `tmp.invalid` where `invalid` does not represent a field or a method name). This will still parse as a `Exp_::Dot` (rather than `Exp_::DotUnresolved`) but with `UnresolvedType` instead of some real one, and we just have to make sure that we process these correctly in the symbolicator.